AutomatticTracks/src/test/java/com/automattic/android/tracks/crashlogging/CrashLoggingTest.kt

```diff
@@ -218,6 +223,44 @@ class CrashLoggingTest {
         assertThat(updatedEvent?.exceptions).contains(DO_NOT_DROP).doesNotContain(TO_DROP)
     }
 
+    @Test
+    fun `should append extra to event before sending it`() {
+        val extraKey = "key"
+        val extraValue = "value"
+
+        initialize(extraKeys = listOf(extraKey), appendBeforeSendAction = { extraValue })
+
+        val updatedEvent = beforeSendModifiedEvent(capturedOptions)
+
+        assertThat(updatedEvent?.getExtra(extraKey)).isEqualTo(extraValue)
+    }
+
+    @Test
+    fun `should not append extra to event if its already there`() {
+        val extraKey = "key"
+        val extraValue = "value"
+        val updatedExtraValue = "updatedValue"
+        val testEventWithExtraApplied = SentryEvent().apply { setExtra(extraKey, extraValue) }
+        initialize(
+            extraKeys = listOf(extraKey),
+            appendBeforeSendAction = { updatedExtraValue }
+        )
+
+        val updatedEvent = beforeSendModifiedEvent(capturedOptions, testEventWithExtraApplied)
+
+        assertThat(updatedEvent?.getExtra(extraKey)).isEqualTo(extraValue)
+    }
+
+    @Test
+    fun `should not modify event if it's not meant to be sent`() {
```

Comment:
🔍  Nitpick: The "not meant to be sent" part might be a bit confusing for newcomers. What if we renamed this to "should not modify events if the user has opted out"?